services/s3/src/main/java/software/amazon/awssdk/services/s3/checksums/ChecksumsEnabledValidator.java

```diff
@@ -131,7 +131,9 @@ public static void validatePutObjectChecksum(PutObjectResponse response, Executi
             byte[] ssHash = Base16Lower.decode(response.eTag().replace("\"", ""));
 
             if (!Arrays.equals(digest, ssHash)) {
-                throw SdkClientException.create("Data read has a different checksum than expected.");
+                throw SdkClientException.create(
+                    String.format("Data read has a different checksum than expected. Was 0x%s, but expected 0x%s",
```

Comment:
Not sure how much the actual values is useful...

Reply:
It was already logging them on the "getobject" case, so I was just making it consistent...